x-pack/plugins/observability_solution/dataset_quality/public/components/flyout/integration_actions_menu.tsx

```diff
@@ -0,0 +1,166 @@
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiContextMenu,
+  EuiContextMenuPanelDescriptor,
+  EuiContextMenuPanelItemDescriptor,
+  EuiPopover,
+} from '@elastic/eui';
+import { css } from '@emotion/react';
+import { Integration } from '../../../common/data_streams_stats/integration';
+import { useDatasetQualityFlyout } from '../../hooks';
+import { useFlyoutIntegrationActions } from '../../hooks/use_flyout_integration_actions';
+import { RouterLinkProps } from './types';
+const seeIntegrationText = i18n.translate('xpack.datasetQuality.flyoutSeeIntegrationActionText', {
+  defaultMessage: 'See integration',
+});
+
+const indexTemplateText = i18n.translate('xpack.datasetQuality.flyoutIndexTemplateActionText', {
+  defaultMessage: 'Index template',
+});
+
+const viewDashboardsText = i18n.translate('xpack.datasetQuality.flyoutViewDashboardsActionText', {
+  defaultMessage: 'View dashboards',
+});
+
+export function IntegrationActionsMenu({ integration }: { integration: Integration }) {
+  const { type, name } = useDatasetQualityFlyout().dataStreamStat!;
+  const { dashboards = [], version, name: integrationName } = integration;
+  const {
+    isOpen,
+    handleCloseMenu,
+    handleToggleMenu,
+    getIntegrationOverviewLinkProps,
+    getIndexManagementLinkProps,
+    getDashboardLinkProps,
+  } = useFlyoutIntegrationActions();
+
+  const actionButton = (
+    <EuiButtonIcon
+      iconType="boxesHorizontal"
+      onClick={handleToggleMenu}
+      data-test-subj="datasetQualityFlyoutIntegrationActionsButton"
+    />
+  );
+
+  const MenuActionItem = ({
+    dataTestSubject,
+    buttonText,
+    routerLinkProps,
+    iconType,
+  }: {
+    dataTestSubject: string;
+    buttonText: string;
+    routerLinkProps: RouterLinkProps;
+    iconType: string;
+  }) => (
+    <EuiButtonEmpty
+      {...routerLinkProps}
+      size="s"
+      css={css`
+        font-weight: normal;
+      `}
+      color="text"
+      iconType={iconType}
+      data-test-subj={dataTestSubject}
+    >
+      {buttonText}
+    </EuiButtonEmpty>
+  );
+
+  const panelItems = useMemo(() => {
+    const firstLevelItems: EuiContextMenuPanelItemDescriptor[] = [
+      {
+        renderItem: () => (
+          <MenuActionItem
+            buttonText={seeIntegrationText}
+            dataTestSubject="datasetQualityFlyoutIntegrationActionOverview"
+            routerLinkProps={getIntegrationOverviewLinkProps(integrationName, version)}
+            iconType="package"
+          />
+        ),
+      },
+      {
+        renderItem: () => (
+          <MenuActionItem
+            buttonText={indexTemplateText}
+            dataTestSubject="datasetQualityFlyoutIntegrationActionTemplate"
+            routerLinkProps={getIndexManagementLinkProps({
+              sectionId: 'data',
+              appId: `index_management/templates/${type}-${name}`,
```

Comment:
could we use deep links here? Maybe it's worth to check`packages/deeplinks/management/deep_links.ts`

Reply:
Why are we the ones sort of creating the url here? I'd expect to be using something like 
```
managementLocator.navigate({ id: 'index_management', type, name })
```